ocp_resources/resource.py

```diff
@@ -1028,6 +1033,58 @@ def wait_for_condition(self, condition: str, status: str, timeout: int = 300) ->
                     if cond["type"] == condition and cond["status"] == status:
                         return
 
+    def wait_for_status_condition(
+        self,
+        wait_condition_fns: List[Callable],
+        not_wait_condition_fns: Optional[List[Callable]] = None,
+        wait_timeout: int = 120,
+        sleep_interval: int = 2,
+    ) -> dict:
+        """
+        Wait for specific status conditions to be met.
+
+        This function continuously checks the current status conditions, waiting for any of the specified "wait"
+        conditions to be satisfied while monitoring for any of the specified "not wait" conditions to trigger a failure.
+
+        Args:
+            wait_condition_fns (List[Callable]): A list of functions that must all return True to indicate that the
+                condition has been met.
+            not_wait_condition_fns (Optional[List[Callable]]):A list of functions where at least one must return True to
+                indicate a failure condition and stop the wait. Default is None.
+            wait_timeout (int, optional): The maximum time to wait for the conditions to be satisfied (in seconds).
+            sleep_interval (int, optional): The interval between checks (in seconds).
+
+        Returns:
+            dict: The condition that indicates the desired status when met.
+
+        Raises:
+            WaitForStatusConditionFailed: If any of the unexpected conditions is met.
+            TimeoutExpiredError: If the timeout expires before the conditions are satisfied.
+        """
+        samples = TimeoutSampler(
+            wait_timeout=wait_timeout, sleep=sleep_interval, func=lambda: self.instance.status.conditions
+        )
+
+        try:
+            for sample in samples:
+                for condition in sample:
+                    if all(wait_fn(condition) for wait_fn in wait_condition_fns):
+                        return condition
```

Comment:
IIUC, this means that this function will return success before checking if an un-wanted condition (i.e. one of the conditions that are covered in the `not_wait_condition_fns` list) is True.

Reply:
Right, why it's a problem?
If the user desired status is met, the unintended status is not relevant.
The user is responsible to give proper condition function without conflicts.